src/expr/src/agg/percentile_cont.rs

```diff
@@ -64,90 +63,75 @@ use crate::Result;
 /// drop table t;
 /// ```
 #[build_aggregate("percentile_cont(float64) -> float64")]
-fn build(agg: &AggCall) -> Result<Box<dyn Aggregator>> {
-    let fraction: Option<f64> = agg.direct_args[0]
+fn build(agg: &AggCall) -> Result<Box<dyn AggregateFunction>> {
+    let fraction = agg.direct_args[0]
         .literal()
         .map(|x| (*x.as_float64()).into());
-    Ok(Box::new(PercentileCont::new(fraction)))
+    Ok(Box::new(PercentileCont { fraction }))
 }
 
-#[derive(Clone, EstimateSize)]
 pub struct PercentileCont {
-    fractions: Option<f64>,
-    data: Vec<f64>,
+    fraction: Option<f64>,
 }
 
-impl PercentileCont {
-    pub fn new(fractions: Option<f64>) -> Self {
-        Self {
-            fractions,
-            data: vec![],
-        }
-    }
+type State = Vec<f64>;
 
-    fn add_datum(&mut self, datum_ref: DatumRef<'_>) {
+impl AggStateDyn for State {}
+
+impl PercentileCont {
+    fn add_datum(&self, state: &mut State, datum_ref: DatumRef<'_>) {
```

Comment:
Ditto

Reply:
This method depends on the direct argument in the function. So I'd like to keep it as it is.